Updated page for Compiler Error CS0116

jmarkman opened the pull request:
## Summary

I tackled these aspects of the page:

* Updated the code sample

Previously, the sample was extremely bare bones and only covered methods existing out of the scope of a class. I added some fields (three variables and an autoproperty), changed the methods to be more concrete concepts (latitude and longitude coordinates rather than Method and Method2), and included a struct. I also placed comments in the code samples pointing out what would make the compiler grumpy and what wouldn't. I also made everything public - while it isn't a Best Practice™, it'll allow those reading/learning to simply copy-paste the code into their editor without running into any accessibility issues.

As the issue actually requested, I added a version of the code sample that would compile. However, the previous sample didn't have a main method. Neither does this one. Both will trigger CS5001. I'm not sure if the snippet should have a main method, I'll leave that for the review of the PR.

* Expanded on the actual text on the page

Just lifted what @BillWagner said in his comment and applied it to the leading short paragraph. However, I also expanded on what would actually happen if the user pasted the compiler-triggering code example into their editor of choice (stuck with Visual Studio because hey, it's docs.***microsoft***.com, right?). I removed the bit about C/C++ programmers because there are plenty of other languages out there where the C# structure is a foreign concept. Scripting languages like Javascript and Python come to mind.

I also added an inline link to the [General Structure of a C# Program](../../../csharp/programming-guide/inside-a-program/general-structure-of-a-csharp-program.md) article even though it was listed in the **See Also** section.

Fixes #9204

Commit:
clarification
- explained where the logging extension method goes
- minor allignment with the LogQuery method

Commit:
applied suggested changes from PR feedback
- removed the *`Extension.cs`* labels
- applied @rpetrusha's suggestions for the LogQuery paragraph

Commit:
Merge branch 'master' of https://github.com/dotnet/docs into jmarkman-update-cs0116

Commit:
updated CS0116
- changed code example to be a bit more realistic/scholastic
- included a struct in the code example
- removed language that singled out C/C++ devs

jmarkman committed:
updated the page
- changed the date
- moved some code around for consistency
- added public modifiers
- updated comments
- specified result of copy-pasting first sample into a code editor (VS)
- added inline link to General Structure of a C# program to text

BillWagner commented on docs/csharp/language-reference/compiler-messages/cs0116.md:
```diff
@@ -9,37 +9,92 @@ ms.assetid: 4cb137b5-ec29-4c1a-adde-9f8424cb9496
-
-This error is often generated by developers from a C/C++ background who forget that in C#, methods and variables must be declared and defined within a struct or class.
-  
+A namespace cannot directly contain members such as fields or methods. However, a namespace can contain other namespaces, structs, and classes. For more information, see the [namespace keyword](../keywords/namespace.md) article.
```
When you split these paragraphs, I'd remove "However" from the start of the next paragraph.

jmarkman replied:
Got it, will fix.

BillWagner commented on docs/csharp/language-reference/compiler-messages/cs0116.md:
```diff
@@ -9,37 +9,92 @@ ms.assetid: 4cb137b5-ec29-4c1a-adde-9f8424cb9496
+        Console.WriteLine($"Lat: {latitude}");
+    }
+
+    // Even if there's a properly defined class and method or a properly defined struct 
```
I'd remove this comment. I think it distracts from the explanation.

BillWagner reviewed (changes requested):
Overall, this is great @jmarkman 

Once you address these comments, it's ready to go.

Thanks for contributing to .NET docs.

Commit:
updated based on feedback from @pkulikov & @billwagner
- split actual error message from rest of article
- removed "even if" comment, leaving comments on actual error-raising code
- followed proper article date format

BillWagner commented:
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

BillWagner commented:
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

BillWagner commented:
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

BillWagner commented:
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.

BillWagner commented:
Thanks for making this changes @jmarkman I'll :shipit: now.

Thanks again for contributing to docs.
